### DISCUSSION TITLE
Nodata and Transparency

### DISCUSSION THREAD

**Discussion by admarrs** [developmentseed/titiler#718] [Q&A] [answered]
I've just started to dip my toes into COGs and titiler so sorry if this is a dumb question. I'm struggling to understand if/how I can get Sentinel-2 nodata areas to render as transparent. 

I'm using rio-cogeo to create a webp encoded COG if Sentinel-2 TCI images then titiler to tile and render. However I get the following for images with partial coverage:
![image](https://github.com/developmentseed/titiler/assets/7544109/594eea4d-08d6-45a0-bbbc-b42ffc89d316)

My question is - how do I turn the black pixels to transparent? I've poured over the rio-cogeo and titiler docs but am still not getting it.

**Comment by vincentsarago** [comment]
@admarrs there is no dumb question! 

The sentinel data you have, have a huge part that should be tagged as Nodata (**0**). Because you are using `Webp` for compression, you'll need to tell rio-cogeo create an alpha band (because Webp or JPEG are not lossless compression, we can't just use the nodata flag, because it will create artefacts) 

Sadly rio-cogeo can't add alpha band at creation time (it can only add mask 😓) so you'll need an extra step (using gdalwarp and the `-dstalpha` option I think) to first create a 4 bands Geotiff  and then use rio-cogeo to create the COG (but at this point you might be better served to create the COG using gdalwarp directly (`-of COG`)

**Reply by admarrs** [reply]
@vincentsarago thank you. Is there a profile other than `Webp` where this will just work?

**Reply by vincentsarago** [reply, marked as answer]
you can try `jpeg` with internal mask 

in rio-cogeo you'll need to use both `--nodata 0` and `--add-mask` option I think

**Reply by admarrs** [reply]
That works but you'll not be surprised that --nodata 0 causes holes in the image because of legitimate pixel values of 0

**Reply by vincentsarago** [reply]
oh because when you rescaled the data you rescaled values to `0` 😓 

wait I may have gotten to fast, do you have a 4 band (RGBA) dataset before creating your COG? if not, this should be easy to create  and then either with the WEBP compression or JPEG (with the --add-mask option) it should work 

if you are able to share your workflow I can maybe help

**Reply by admarrs** [reply]
My workflow is using the sentinelsat api to search for then download the _TCI jp2 files then create the cog using:

```
def _translate(src_path, dst_path, profile="jpeg", profile_options={}, **options):
    """Convert image to COG."""
    # Format creation option (see gdalwarp `-co` option)
    output_profile = cog_profiles.get(profile)
    output_profile.update(dict(BIGTIFF="IF_SAFER"))
    output_profile.update(profile_options)

    # Dataset Open option (see gdalwarp `-oo` option)
    config = dict(
        GDAL_NUM_THREADS="ALL_CPUS",
        GDAL_TIFF_INTERNAL_MASK=True,
        GDAL_TIFF_OVR_BLOCKSIZE="128",
    )

    cog_translate(
        src_path,
        dst_path,
        output_profile,
        config=config,
        in_memory=False,
        nodata=0,
        add_mask=True,
        quiet=True,
        **options,
    )
    return True
```

where `profile_options` and `options` are just the defaults. The dataset is just standard Sentinel-2 3 band RGB with 0 as the nodata value

**Reply by vincentsarago** [reply]
The sentinel-2 are also available as COG publicly available on AWS

https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/W/XA/2023/10/S2B_36WXA_20231027_0_L2A/TCI.tif
